pycsw/ogc/api/records.py

```diff
@@ -1048,22 +1048,70 @@ def record2json(record, url, collection, stac_item=False):
 
     if record.links:
         rdl = record_dict['links']
-
         for link in jsonify_links(record.links):
-            link2 = {
-                'href': link['url'],
-                'name': link['name'],
+            link2 = {   
+                'type': link['protocol'],
                 'description': link['description'],
-                'type': link['protocol']
-            }
-            if 'rel' in link:
-                link2['rel'] = link['rel']
-            elif link['protocol'] == 'WWW:LINK-1.0-http--image-thumbnail':
-                link2['rel'] = 'preview'
-            elif 'function' in link:
-                link2['rel'] = link['function']
-
-            rdl.append(link2)
+                'title': link['name'],
+                'href': link['url']
+                }
+            if link['protocol']:
+                if 'OGC:WMS' in link['protocol'].upper():
+                    link2['rel'] = 'map'
+                    link2['templated'] = 'true'
+                    # assumes link['url'] includes '&layers=...', else link['name'] contains layername(s) 
```

Comment:
alternative would be to use owslib.wms130.__build_getmap_request (but challenge to replace bbox array for `{bbox}`)